target/linux/ath79/dts/qca9531_glinet_gl-s200.dtsi

```diff
@@ -0,0 +1,182 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca953x.dtsi"
+
+/ {
+	compatible = "glinet,gl-s200", "qca,qca9531";
+	model = "GL.iNet GL-S200";
+
+	aliases {
+		label-mac-device = &eth0;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+
+		poll-interval = <20>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			linux,input-type = <EV_KEY>;
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+
+		switch {
+			label = "switch";
+			linux,code = <KEY_SETUP>;
+			linux,input-type = <EV_SW>;
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		blue_led {
+			label = "gl-s200:blue";
+			gpios = <&gpio 13 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		white_led {
+			label = "gl-s200:green";
+			gpios = <&gpio 12 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+
+		orange_led {
+			label = "gl-s200:orange";
+			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "off";
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+
+		gpio_zigbee_reset {
+			gpio-export,name = "gpio1";
+			gpio-export,output = <1>;
+			gpio-export,direction_may_change;
+			gpios = <&gpio 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_zwave_reset {
+			gpio-export,name = "gpio2";
+			gpio-export,output = <1>;
+			gpio-export,direction_may_change;
+			gpios = <&gpio 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_zigbee_upgrade {
+			gpio-export,name = "gpio11";
+			gpio-export,output = <1>;
+			gpio-export,direction_may_change;
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio_zwave_upgrade {
+			gpio-export,name = "gpio17";
+			gpio-export,output = <1>;
+			gpio-export,direction_may_change;
+			gpios = <&gpio 17 GPIO_ACTIVE_HIGH>;
+		};
+	 };
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+	num-cs = <2>;
+	cs-gpios = <0>, <0>;
```

Comment:
As far as I know, this driver does not use them (guessing the pinctrl is set by bootloader), but we want the DTS to be an accurate definition of the hardware.
If you are using GPIOs for chip selects (I believe you need to for this SoC, no native pins, but GPIO5 by default is CS0?), could you please define the GPIO numbers and settings here?

Reply:
already edited

Reply:
Do want the `num-cs` and `cs-gpios` here, but with accurate GPIO information:
Example (change 5 & 31 to suit, (provide link to uboot source, and I could check))?
```
	num-cs = <2>;
	cs-gpios = <&gpio 5 GPIO_ACTIVE_LOW>, <&gpio 31 GPIO_ACTIVE_LOW>;
```

Reply:
There is no u-boot support for ath79 in the original openwrt.